### DISCUSSION TITLE
[Request] Kasm Workspaces

### DISCUSSION THREAD

**Discussion by faqall** [tteck/Proxmox#338] [Ideas]
Kasm Workspaces is a container streaming platform for delivering browser, desktop and application workloads to the web browser

https://www.kasmweb.com/index.html

**Comment by mattague** [comment]
I've been really interested in setting up kasm, checked to see if there was anything helpful here, guess i will have to do it manually for now.

**Reply by tteck** [reply]
I created one some time ago, It's resource hungry and forever to install.
Default resources looked like 2vCPU 4GB RAM 4GB swap 60GB storage and these are min.

**Reply by tteck** [reply]
You can create a Debian LXC with the above resources, then:
```
cd /tmp
wget -q https://kasm-static-content.s3.amazonaws.com/kasm_release_1.12.0.d4fd8a.tar.gz
tar -xf kasm_release*.tar.gz
bash kasm_release/install.sh --admin-password changeme --user-password changeme -L 8443 -e
```
and then wait...

(https) IP:8443

Login:
admin@kasm.local - changeme
user@kasm.local - changeme

**Reply by damadorPL** [reply]
https://www.youtube.com/watch?v=fzYh86fz1iY  here  is docker way

**Reply by second2falcon** [reply]
@tteck I was just looking into Kasm again, but the specs you listed (2vCPU 4GB RAM 4GB swap 60GB storage) are the minimum for the workspace aka VM and not the server / VM Kasm is running itself in. The actual recommended spec for minimum workspace spec and max 8 workspaces is 16 CPUs, 64GB RAM. Probably close to or more than what some homelabbers have at capacity limit.
I'll experiment the coming time what a good spec it where the workspaces won't lag too much. It might take a while cause I need to make some room on one of my nodes (most are close to RAM limit) for Kasm.

Edit: I know for the standalone server it mentions a lot lower requirements, but what is the point of having lower specs with everything combined than distributed.

Also, [source](https://www.kasmweb.com/docs/latest/how_to/sizing_operations.html#id3)

**Comment by stephenc01** [comment]
Hey, just brining this back up I do use it on my cluster a script here would be nice. It would just be an extension of the docker script to install the additional package in LCX

**Comment by GrumpyMeow** [comment]
Please be aware that setting up Kasm in a container might fail when your host is using ZFS storage.
[Thread on proxmox forum](https://forum.proxmox.com/threads/docker-failed-to-register-layer-applylayer-exit-status-1-stdout-stderr-unlinkat-var-log-apt-invalid-argument.119954/page-2)
I had Kasm working on my previous system (BTRFS), but after migrating to my new system (ZFS) it didn´t work any more.
